### PR TITLE
DS-6: 4 or 6 bits per char

### PR DESCRIPTION
**Pull request by evanfields**
Previously we only supported `bits_per_char = 2` for encoding/decoding. Now we also support 4 or 6 bits per character, just like the Python library. Tests are updated to match.

This work is necessary but not sufficient for ticket DS-6.

### CODE REVIEW

File: src/GeohashHilbert.jl [hunk 1]
```diff
@@ -44,19 +44,61 @@ const HILBERT_RECURSION = dict_to_idict(Dict(
...
+
+"Encode an integer using 6 bits per character; return a string of the specified number of
+characters. `nchar` must be large enough to encode `x`: ``x < 64^nchar``."
+function base64_encode_int(x, nchar, charlist = BASE64_CHARS)
+    chars = fill('0', nchar)
+    for i in nchar : -1 : 1
+        x, rem = divrem(x, 64)
+        chars[i] = charlist[rem + 1]
+    end
+    return String(chars)
```
Comment (nshi) [inline src/GeohashHilbert.jl:59]: The indentation in this function seems to be different from the rest of this file.

File: src/GeohashHilbert.jl [hunk 2]
```diff
@@ -44,19 +44,61 @@ const HILBERT_RECURSION = dict_to_idict(Dict(
...
+end
+"Decode a base 64 encoded string back into an integer. This function will error if
```
Comment (nshi) [inline src/GeohashHilbert.jl:62]: Do you want to have a newline between functions?

File: test/runtests.jl [hunk 3]
```diff
@@ -82,6 +83,10 @@ function test_match_python_encode(bits_per_char = 2)
 end
 
 Random.seed!(47)
-test_cell_centers_encode_decode()
```
Comment (nshi) [inline test/runtests.jl:85]: Do you no longer need the test?

File: src/GeohashHilbert.jl (hunk 1 above)
Comment (evanfields) [reply src/GeohashHilbert.jl:59]: This file is mixed tabs and spaces. I started writing it when I was still pro-tab, and continued writing after I came to believe that the war was over, the good guys lost, but there's benefit to conformity so we should all give up and use spaces. 

I think the least painful way to resolve the whitespace and indentation is to land #2, merge master into here, and have a whitespace-resolution commit.

File: test/runtests.jl (hunk 3 above)
Comment (evanfields) [reply test/runtests.jl:85]: We now call it with different values for `bits_per_char`, though I had sloppily neglected that below (line 89) - fixed.

**Review by nshi — APPROVED** [review]

**Pull request merged**